pages/_app.tsx

```diff
@@ -58,16 +67,33 @@ export default function App({ Component, pageProps }: AppProps) {
       }
     })();
   }, [state, dispatch]);
+
   return (
     <AppStateContext.Provider value={state}>
       <AppDispatchContext.Provider value={dispatch}>
         <div className="relative min-h-screen">
           <div id="modal" />
           <NavBar />
-          <div className="pt-20 pb-28">
-            <Component {...pageProps} />
-            <Footer />
+          {!!state.address && Object.entries(state.contracts).length > 0 && (
+            <Sidebar isOpen={hasSidebar} onClose={() => setHasSidebar(false)} />
+          )}
+          <div className="pt-20 pb-28 md:pl-72">
+            <button
+              className="ml-4 mt-4 flex items-center space-x-2 text-zinc-500 md:hidden"
+              onClick={() => {
+                setHasSidebar(true);
+              }}
+            >
+              <span className="text-xs">Open sidebar</span>
+              <ArrowRightIcon className="h-4 w-4" />
+            </button>
+            {path === "/" && !!state.address && !!state.currentContract ? (
+              <Proposals />
+            ) : (
+              <Component {...pageProps} />
+            )}
           </div>
+          <Footer />
```

Comment:
I think it will look nice if the layout of `Footer` align with `Proposals` or `Component`. It looks a bit weird that the texts of footer isn't in the middle.